src/domain/usecases/CheckSettingsPermissionsUseCase.ts

```diff
@@ -1,15 +1,17 @@
 import _ from "lodash";
+
 import { User } from "../../data/entities/User";
 import { UseCase } from "../../webapp/CompositionRoot";
 import { NamedRef } from "../entities/Ref";
 import { ConfigRepository } from "../repositories/ConfigRepository";
+import { Config } from "../entities/Config";
 
 export class CheckSettingsPermissionsUseCase implements UseCase {
     constructor(private configRepository: ConfigRepository) {}
 
-    public async execute(): Promise<boolean> {
+    public async execute(config: Config): Promise<boolean> {
         const user = await this.configRepository.getUser();
-        const permissions = await this.configRepository.getSettingsPermissions();
+        const { settingsPermissions: permissions } = config ?? (await this.configRepository.get());
```

Comment:
The `(await this.configRepository.get())` won't be reached right? as config will be always defined